File: Regular Expressions/Code/chapter6/6-22/recipe6-22.pl
#!/usr/bin/perl -w
use strict;

my $fn = $ARGV[0] || die "Please supply a paramter\n";

open( FILE, $fn ) || die "Could not open file\n";

while ( <FILE> )
{ 
	print $_ if ( s/^([^=]+?)=(.*)$/Key: '$1'; Value: '$2'/ );
}

close( FILE );

File: Regular Expressions/Code/chapter6/6-22/recipe6-22.py
#!/usr/bin/python
import re
import sys

nargs = len(sys.argv)

if nargs > 1:

	myfile = sys.argv[1]
	output = open( myfile )
	regex = re.compile( r'^([^=]+?)=(.*)' )

	lines = output.readlines()
	for line in lines:
		if regex.match( line ):
			formatted = regex.sub( r'Value is "\2" for key "\1"', line )
			print formatted,

	output.close()

else:
	print 'Please supply a parameter!',
